decorators.py

```diff
@@ -0,0 +1,164 @@
+from functools import wraps
+from io import BytesIO
+import gzip
+
+from flask import jsonify, Response, g, request
+from flask_sqlalchemy_session import current_session
+
+from model import Library
+from util import GeometryUtility
+from util.flask_util import originating_ip
+from util.geo import Location, InvalidLocationException
+from util.problem_detail import ProblemDetail
+from problem_details import LIBRARY_NOT_FOUND
+
+
+def deprecated_route(f):
+    """Report usage of a deprecated route"""
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        # TODO: Log the usage of a deprecated route and emit a metric
+
+        return f(*args, **kwargs)
+
+    return decorated
+
+
+def uses_location(f):
```

Comment:
Decorator that looks for location, tries to store it in `g.location` as a `Location` instance.